packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import DebounceInput from 'react-debounce-input';
+import getMinutes from 'date-fns/get_minutes';
+import getHours from 'date-fns/get_hours';
+import getDate from 'date-fns/get_date';
+import setDate from 'date-fns/set_date';
+import setMinutes from 'date-fns/set_minutes';
+import lastDayOfMonth from 'date-fns/last_day_of_month';
+import isSameMinute from 'date-fns/is_same_minute';
+import startOfDay from 'date-fns/start_of_day';
+import startOfMinute from 'date-fns/start_of_minute';
+import format from 'date-fns/format';
+import twoDigits from '../shared/utils/format/twoDigits';
+import DateTimePicker from '../DateTimePicker';
+import theme from './InputDateTimePicker.scss';
+
+const DEBOUNCE_TIMEOUT = 300;
+
+const splitDateAndTimePartsRegex = new RegExp(/^\s*([^\s]+?)\s+([^\s]+?)\s*$/);
+const datePartRegex = new RegExp(/^\s*([0-9]{4})-([0-9]{1,2})-([0-9]{1,2})\s*$/);
+const timePartRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
+
+function hoursAndMinutesToTime(hours, minutes) {
+	return hours * 60 + minutes;
+}
+
+function getTextDate(date, time) {
+	if (date === undefined) {
+		return '';
+	}
+
+	const dateText = format(date, 'YYYY-MM-DD');
+
+	if (time === undefined) {
+		return dateText;
+	}
+
+	const hours = Math.floor(time / 60);
+	const minutes = time % 60;
+
+	const timeText = `${twoDigits(hours)}:${twoDigits(minutes)}`;
+
+	return `${dateText} ${timeText}`;
+}
+
+function extractDate(strToParse) {
+	const dateMatches = strToParse.match(datePartRegex);
+
+	if (!dateMatches) {
+		const errMsg = 'DATE - INCORRECT FORMAT';
```

Comment:
Useless constant.

Reply:
Not necessary but far to be useless.
It improves readability by keeping the return statement simple, add semantic of what's returning as second value.
Then for i18n, we'll need to add some more stuff, this will definitely don't feat with cleanliness.